src/main/java/com/worksap/nlp/sudachi/dictionary/DoubleArrayLexicon.java

```diff
@@ -167,4 +175,9 @@ public void calculateCost(Tokenizer tokenizer) {
             wordParams.setCost(wordId, (short) cost);
         }
     }
+
+    public void setDictionaryId(int id) {
```

Comment:
I want to ultimately remove the Lexicon interface. I am not sure that it is correct that single lexicon (mostly an implementation detail) and a set of them (can be an API) share an interface.

Reply:
I wanted to separate the APIs for searchable vocabulary from ones for binary images, so I created `Lexicon`. If you understand such a thing, there is no problem with this change.

Reply:
So the intent is to have plain binary dictionaries non-searchable to user? They have to create lexicon from a collection of binary dictionaries to make them searchable?

Reply:
The dictionary class tends to have more APIs, and it seemed complicated to have different layers mixed together, so I separated `BinaryDictionary` as an entity and `Lexicon` as a functionality provider. If you decide you don't need that distinction, you can do away with `Lexicon`.

Reply:
I was not clear enough, I would like to eventually remove Lexicon interface because it serves no purpose in my opinion. 

Binary dictionary represents a in-memory compiled or loaded from disk binary dictionary, it exposes headers and internal structures, but can't be used for lookup. DoubleArrayLexicon (may be we should rename it) is a single dictionary internal lookup API and probably should not be exposed to users at all. LexiconSet is a collection of a system + list of dependent user lexicons and probably should be the only exposed lookup API. 

`Lexicon` interface here in my opinion is not a useful abstraction here and instead of mocking (if somebody wanted to do that) it would be better to create a lightweight in-memory compiled dictionary with words they want to use, as it is rather easy now.

Reply:
Well, yes, Lexicon was structured more for me to organize code than for users, so it may not be necessary now. I agree that it would be better to have another interface for users to create in-memory, lightweight dictionaries.